kornia/geometry/conversions.py

```diff
@@ -146,40 +150,54 @@ def angle_axis_to_rotation_matrix(angle_axis: torch.Tensor) -> torch.Tensor:
             "Input size must be a (*, 3) tensor. Got {}".format(
                 angle_axis.shape))
 
-    def _compute_rotation_matrix(angle_axis, theta2, eps=1e-6):
+    num_dim: int = len(angle_axis.shape)
+    if num_dim == 1:
+        angle_axis = torch.unsqueeze(angle_axis, dim=0)
+
+    # this is batched dot product
+    #theta2 = torch.einsum('ij, ij->i', angle_axis, angle_axis).unsqueeze(1)
+    theta2 = torch.sum(angle_axis * angle_axis, dim=-1, keepdim=True)
+
+    def rot_mat(r00, r01, r02, r10, r11, r12, r20, r21, r22) -> torch.Tensor:
+        rotation_matrix: torch.Tensor = torch.cat(
+            [r00, r01, r02, r10, r11, r12, r20, r21, r22], dim=-1)
+        return rotation_matrix.view(-1, 3, 3)
+
+    def cond_1() -> torch.Tensor:
         # We want to be careful to only evaluate the square root if the
         # norm of the angle_axis vector is greater than zero. Otherwise
         # we get a division by zero.
-        k_one = 1.0
         theta = torch.sqrt(theta2)
+        k_one = torch.ones_like(theta)
+
         wxyz = angle_axis / (theta + eps)
-        wx, wy, wz = torch.chunk(wxyz, 3, dim=1)
+        wx, wy, wz = torch.chunk(wxyz, 3, dim=-1)
+
         cos_theta = torch.cos(theta)
         sin_theta = torch.sin(theta)
 
         r00 = cos_theta + wx * wx * (k_one - cos_theta)
-        r10 = wz * sin_theta + wx * wy * (k_one - cos_theta)
-        r20 = -wy * sin_theta + wx * wz * (k_one - cos_theta)
-        r01 = wx * wy * (k_one - cos_theta) - wz * sin_theta
+        r01 = wz * sin_theta + wx * wy * (k_one - cos_theta)
```

Comment:
Why was the matrix transposed?  Based on results I am seeing I think the original way was correct.

Reply:
was trying another implementation I found, however, the tests I provide above doesn't work. Even with the original implementation. So wrong either in the implementation or numerical

Reply:
any help is welcomed